ocaml/xapi/bios_strings.ml

```diff
@@ -23,35 +23,117 @@ let remove_invisible str =
   let str = String.concat "\n" l in
   String.fold_left (fun s c -> if c >= ' ' && c <= '~' then s ^ (String.of_char c) else s) "" str
```

Comment:
This is pretty inefficient. If we have `Astring` available, `Astring.String.filter` would be a good choice.

Reply:
I'll do that.

I think most uses of this function can be dropped and just be used for the HP special case. I'll double check if dmidecode can return invisible characters.

Reply:
I'd also like to see the output in the special case, I suspect the filtering of "#-commented" lines comes from trying to process dmidecode output.

Reply:
In the HP special case only 6 bytes get read, all the processing and trimming done is unnecessary.